Hold-Krykke/Security_and_FullstackJavascript_Exam
# SSO OIDC

# How to Review

1. Checkout Branch SSO-OIDC
2. `npm install` in Root of "FindYourFriends" folder.
3. `npm run start` 
4. Open app in your Expo on Mobile. 
5. Try Google Login
6. Read code in [`./screens/LoginScreen.js`](https://github.com/Hold-Krykke/Security_and_FullstackJavascript_Exam/pull/10/files)
7. Read Documentation at [Google - Expo Documentation](https://docs.expo.io/versions/latest/sdk/google/)
8. Report back here with a Review! 
    1. Did it work on your device? 
    2. Any questions? (I highly recommend watching the two videoes first, though.)
    3. Anything I should comment in the code? 

### Discussion / Next Steps
1. I strongly recommend watching the two videoes for understanding! 
    1. [First one](https://youtu.be/t18YB3xDfXI)! 
    2. [Second one](https://youtu.be/996OiexHze0)! 
2. Discuss how we send the idToken / accessToken back to the backend. 
3. Right now we do it all on the front channel. This isn't the most secure solution. The best way is to get a code, which the backend can exchange for an idToken. How do we do that? 

### Progress
~~[Expo Docs](https://docs.expo.io/versions/latest/sdk/app-auth/)~~
~~[OAuth App](https://github.com/organizations/Hold-Krykke/settings/applications/1293630)~~
Det virker på min telefon! Har brugt [denne dokumentation. ](https://docs.expo.io/versions/latest/sdk/google/)

### Other
Key used was made and taken from [here](https://console.developers.google.com/apis/credentials?project=securityexam).
